src/_guides/language/effective-dart/design.md

```diff
@@ -658,50 +696,121 @@ can.
 
 ### DO use getters for operations that conceptually access properties.
 
-If the name of the method starts with `get` or is a noun phrase like `length` or
-`size` that's a sign you're better off using a getter. You
-should define a getter instead of a method when all of these are true:
+Deciding when a member should be a getter versus a method is a challenging,
+subtle, but important part of good API design, hence this very long guideline.
+Some other language's cultures shy away from getters. They only use them when
+the operation is almost exactly like a field — it does a miniscule amount of
+calculation on state that lives entirely on the object. Anything more complex or
+heavyweight than that gets `()` after the name to signal "computation goin' on
+here!" because a bare name after a `.` means "field".
+
+Dart is *not* like that. In Dart, *all* dotted names are member invocations that
+may do computation. Fields are the special-case — getters whose implementation
+is provided by the language. In other words, getters are not "particularly slow
+fields" in Dart, fields are "particularly fast getters".
```

Comment:
, -> ; (if you don't hate semicolons)
or even , -> :
The "," made this a little hard to parse initially.

Reply:
Generally dislike ";", but I think it's best here.